### PR TITLE
[Spritelab] Close WorkspaceAlert for execution error on reset

### PR DESCRIPTION
**Pull request by maddiedierker**
This is a minor update to how we alert users about execution errors in Spritelab. We weren't removing executionError alerts, so alerts began stacking. Now, every time you run or reset the program (e.g., the program state is reset), we clear the previous executionError alert if one exists.

## Links

- [STAR-2165](https://codedotorg.atlassian.net/browse/STAR-2165)

### CODE REVIEW

**Comment by maddiedierker** [comment]
hold off on reviewing -- i'm going to make some functional changes and fix some tests. i'll re-tag the team when it's ready!

**Comment by maddiedierker** [comment]
ready for review! @mikeharv thanks for helping me work through this this morning -- could you take a look?

File: apps/src/p5lab/spritelab/SpriteLab.js
Comment (mikeharv) [inline apps/src/p5lab/spritelab/SpriteLab.js]: Is this comment still accurate? It seems like we do display the alert before pressing Run as well. Does our auto-preview state count as running?

File: apps/src/StudioApp.js
Comment (mikeharv) [inline apps/src/StudioApp.js:3173]: I like this! It makes everything above much easier to read.

**Review by mikeharv — APPROVED** [review]
Pulled and test thoroughly. Everything seems to be working just how I'd expect it to. Thanks for making these changes. Before merging, could you just double-check that the comment I flagged matches the new behavior?

File: apps/src/p5lab/spritelab/SpriteLab.js
Comment (maddiedierker) [reply apps/src/p5lab/spritelab/SpriteLab.js]: oh no, i updated this comment before you and i chatted and i fixed the alert behavior. i'll revert this -- thank you for catching!

Comment (mikeharv) [reply apps/src/p5lab/spritelab/SpriteLab.js]: Okay, that's honestly what I was figuring! Thanks

**Pull request merged**
